File: VOZ DO NARRADOR/NARRADOR.py
import os
import telegram
from telegram.ext import Updater, MessageHandler, Filters
from google.cloud import texttospeech

# Configuração do bot
bot_token = 'seu_token_aqui'
bot = telegram.Bot(token=bot_token)

# Configuração do Google Cloud Text-to-Speech
os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = 'CAMINHO_DO_ARQUIVO_DE_CREDENCIAIS_AQUI.json'
client = texttospeech.TextToSpeechClient()

# Função para converter texto em fala
def text_to_speech(text, voice_name):
    input_text = texttospeech.SynthesisInput(text=text)
    voice = texttospeech.VoiceSelectionParams(language_code='pt-BR', name=voice_name)
    audio_config = texttospeech.AudioConfig(audio_encoding=texttospeech.AudioEncoding.MP3)
    response = client.synthesize_speech(input=input_text, voice=voice, audio_config=audio_config)
    return response.audio_content

# Função para receber mensagens dos usuários
def handle_message(update, context):
    # Obtém o texto da mensagem enviada pelo usuário
    text = update.message.text
    # Converte o texto em fala com a voz do Ricardo
    audio_data = text_to_speech(text, 'pt-BR-Wavenet-B')
    # Envia o áudio como mensagem de voz para o usuário
    update.message.reply_voice(voice=audio_data)

# Criação do objeto Updater e do dispatcher
updater = Updater(token=bot_token, use_context=True)
dispatcher = updater.dispatcher

# Registra o handler para mensagens de texto
dispatcher.add_handler(MessageHandler(Filters.text, handle_message))

# Inicia o bot
updater.start_polling()
updater.idle()

# Para criar um bot do Telegram em Python que permite enviar um texto e receber um áudio com a voz do Ricardo ou Felipe, podemos utilizar a API do Google Cloud Text-to-Speech, que permite converter texto em fala.

# Antes de começar, é necessário criar uma conta no Google Cloud e configurar as credenciais de acesso à API. Em seguida, vamos utilizar a biblioteca python-telegram-bot para criar o bot e receber as mensagens dos usuários.

# No código acima, a função text_to_speech recebe um texto e o nome da voz que deve ser usada na conversão para fala. Neste exemplo, estamos utilizando as vozes "pt-BR-Wavenet-B" e "pt-BR-Wavenet-F", que correspondem às vozes do Ricardo e do Felipe, respectivamente. Você pode conferir outras vozes disponíveis na documentação da API do Google Cloud Text-to-Speech.

# A função handle_message é registrada como um handler para mensagens de texto e é executada sempre que o bot recebe uma mensagem de texto de um usuário. Essa função converte o texto em fala usando a voz do Ricardo e envia o áudio como mensagem de voz para o usuário.

# Para utilizar a voz do Felipe, basta alterar o segundo parâmetro da função text_to_speech para 'pt-BR-Wavenet-F'.

File: VOZ DO NARRADOR/NARRADOR.js
const TelegramBot = require('node-telegram-bot-api');
const textToSpeech = require('@google-cloud/text-to-speech');
const fs = require('fs');

// Configuração do bot
const botToken = 'seu_token_aqui';
const bot = new TelegramBot(botToken, { polling: true });

// Configuração do Google Cloud Text-to-Speech
const credentialsPath = 'CAMINHO_DO_ARQUIVO_DE_CREDENCIAIS_AQUI.json';
const client = new textToSpeech.TextToSpeechClient({ keyFilename: credentialsPath });

// Função para converter texto em fala
async function textToSpeechHandler(text, voiceName) {
  const request = {
    input: { text: text },
    voice: { languageCode: 'pt-BR', name: voiceName },
    audioConfig: { audioEncoding: 'MP3' },
  };
  const [response] = await client.synthesizeSpeech(request);
  return response.audioContent;
}

// Função para receber mensagens dos usuários
bot.on('message', async (msg) => {
  const chatId = msg.chat.id;
  const text = msg.text;
  const audioData = await textToSpeechHandler(text, 'pt-BR-Wavenet-B');
  bot.sendVoice(chatId, audioData);
});

// Para criar um bot do Telegram em JavaScript que permite enviar um texto e receber um áudio com a voz do Ricardo ou Felipe, podemos utilizar a API do Google Cloud Text-to-Speech, que permite converter texto em fala.

// Antes de começar, é necessário criar uma conta no Google Cloud e configurar as credenciais de acesso à API. Em seguida, vamos utilizar a biblioteca node-telegram-bot-api para criar o bot e receber as mensagens dos usuários.

// No código acima, a função textToSpeechHandler recebe um texto e o nome da voz que deve ser usada na conversão para fala. Neste exemplo, estamos utilizando as vozes "pt-BR-Wavenet-B" e "pt-BR-Wavenet-F", que correspondem às vozes do Ricardo e do Felipe, respectivamente. Você pode conferir outras vozes disponíveis na documentação da API do Google Cloud Text-to-Speech.

// A função bot.on('message', ...), registrada como um handler para mensagens, é executada sempre que o bot recebe uma mensagem de texto de um usuário. Essa função converte o texto em fala usando a voz do Ricardo e envia o áudio como mensagem de voz para o usuário.

// Para utilizar a voz do Felipe, basta alterar o segundo parâmetro da função textToSpeechHandler para 'pt-BR-Wavenet-F'.
